app/controllers/architectures_controller.rb

```diff
@@ -39,4 +55,33 @@ def destroy
       process_error
     end
   end
+
+  private
+
+  def controller_permission
+    'architectures'
+  end
+
+  def columns
+    [js_sortable_col(_('Name'), 'name'), _('Operating systems'), _('Hosts')]
+  end
+
+  def rows(architectures)
+    architectures.map do |arch|
+      name = js_link_if_can_edit(arch, edit_architecture_path(:id => arch), arch.name)
+      os = arch.operatingsystems.map(&:to_label).to_sentence
+      hosts_count = js_link_to(hosts_count(:architecture)[arch], hosts_path(:search => "architecture = #{arch}"))
```

Comment:
Should `hosts_count(:architecture)` be stored outside of the loop? Otherwise I'm afraid it may calculate the value for every architecture. Even if it comes from a cache, that's still overhead.